Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/DocumentAdapter.h

```diff
@@ -60,10 +62,37 @@ namespace AZ::DocumentPropertyEditor
         //! \see RoutingAdapter
         virtual void SetRouter(RoutingAdapter* router, const Dom::Path& route);
 
+        //! If true, debug mode is enabled for all DocumentAdapters.
+        //! \see SetDebugModeEnabled
+        static bool IsDebugModeEnabled();
+        //! Enables or disables debug mode globally for all DocumentAdapters.
+        //! Debug mode adds expensive extra steps to notification operations to ensure the
+        //! adapter is behaving correctly and will report warnings if an issue is detected.
+        //! This can also be set at runtime with the `ed_debugDocumentPropertyEditorUpdates` CVar.
+        static void SetDebugModeEnabled(bool enableDebugMode);
```

Comment:
should this be an override-able method that the child adapters can override instead? I can easily see a use case where systems would want to debug just their adapter instead of all adapters.

Reply:
Debug mode will only log something if an adapter malfunctions in some way, so I'm not expecting the need for granular, per-adapter debugging. Instead the intent is this either gets triggered from unit tests (using this method) or from the Editor by just setting the `ed_debugDocumentPropertyEditorUpdates` CVar (which you can do prior to pulling up a DPE with your adapter) - I'm not expecting any other code-side changes to enable debug mode, but I can re-evaluate once we have more adapters in and we've worked through debugging them firsthand.